lib/TH/TH.h

```diff
@@ -3,6 +3,10 @@
 
 #include "THGeneral.h"
 
+#ifdef TH_GENERIC_USE_HALF
```

Comment:
This was supposed to be '#if TH_GENERIC_USE_HALF', fixed. 
The goal with cutorch is to avoid half<->float conversions when not necessary. 
This PR can be committed w/o cutorch and no module that do not define TH_GENERIC_USE_HALF would be affected - cutorch woudl still be using its own Half, too.
I just rebuilt and ran the test with this branch and cutorch master.